### PR TITLE
Define katello for satellite and orcharhino

### PR DESCRIPTION
**Pull request by ekohl**
This allows ifdef::katello for Katello-plugin-only installs while still including those in Satellite and Orcharhino builds. It also allows ifndef::katello for vanilla Foreman installations.

Right now this PR is incomplete, but I want to submit it for consideration just as an idea (rather than the exact implementation). What do you think?

### CODE REVIEW

**Comment by ehelms** [comment]
If the intent is to try to say Katello is a subset of Satellite, I think that is an incorrect approach as its not strictly true and could lead to confusing situations. I don't entirely grasp what the problem being encountered is so I can only try to give that thought for now.

**Comment by ekohl** [comment]
> If the intent is to try to say Katello is a subset of Satellite, I think that is an incorrect approach as its not strictly true and could lead to confusing situations. I don't entirely grasp what the problem being encountered is so I can only try to give that thought for now.

I think that where it isn't true, we should have additional `ifdef::satellite[]` or `ifndef::satellite[]`. More of an opt-out than opt-in. The problem I encountered is that in https://github.com/theforeman/foreman-documentation/pull/679 I end up with a **lot** of `ifdef::katello`.

**Review by melcorr — COMMENTED** [review]
I understand what you're trying to do. 
I've been sitting here for a while going around in circles with this. 
I'm trying to think from all perspectives; katello ifdefs are frequent at the moment. As we expand the docset `ifndefs` for satellite will be as prevalent if we introduce this, and definitely if ATIX add their Katello content here, which I would love.
If I weigh the consequences of this, we come out with an equal amount of conditionalising in the long term, just it's `satellite` rather than `katello`. 
I do think that the change makes it more confusing than the declaration of the relevant scenarios in the ifdef. When I say this, I am thinking about the level of contributions we've had from people less familiar with Foreman over the past few months who seem to genuinely struggle. We've new people joining at some point in the near future, allegedly. 
I also think that there is some merit to considering all our potential scenarios in a conscious way when declaring the ifdef. I know that is not how engineers view the world, but there's a merit in being forced to think about who precisely will be reading it. 
All that said, I am not against this. 
I'd really like to hear opinions from @lzap  and @maximiliankolb and we'll make a decision. 
I hope that's OK. 
PS: If we do merge this, we'll need to add this to the readme.

**Comment by melcorr** [comment]
Looking at the errors in the build, this also breaks our linking conventions?

**Comment by melcorr** [comment]
In light of your comment on #676 @maximiliankolb and since this breaks how we build our links, I'm leaning towards thinking this might bring about more problems than it solves.

**Review by maximiliankolb — APPROVED** [review]
LGTM.

**Comment by maximiliankolb** [comment]
> In light of your comment on #676 @maximiliankolb and since this breaks how we build our links, I'm leaning towards thinking this might bring about more problems than it solves.

Your call. For us, the only change would be to add `:katello:` to each document using upstream files, so that's ok for me. As both orcharhino and Satellite uses Katello, I think it's fairly obvious where I would need to make adjustments/ add `,orcharhino` to `ifdef::satellite[]`.

(I've reviewed this before I saw your comment; the changes look good to me.)

**Comment by melcorr** [comment]
Will it be easy to get around the link building problems that are now occurring because of this change?

**Comment by lzap** [comment]
Let me recap what we have today. It all started when we created two build flags: `foreman` and `satellite`. In the beginning, everything was very clear - it is either foreman-core or satellite/katello. As we were adding more and more content and Katello project started using this repo for upstream docs, we have ended up with the following build attributes:

* foreman
* foreman-el
* foreman-deb
* katello
* satellite
* orcharhino

These are disjoint groups, except two cases (let me use `<` for "includes")

* foreman-el < foreman
* foreman-deb < foreman

This only exists so its easier to type `ifdef::foreman[]` instead of `ifdef::foreman-el,foreman-deb[]`. Ewoud suggest to create a new one:

* satellite < katello

I am trying to imagine what happens when we get to the point when Katello becomes really a plugin that can be installed on top of Foreman. We cannot do this anymore:

* satellite < katello < foreman

Why exactly you need this when you can use either `ifdef::katello,satellite[]` or one or the other?

I am not entirely sure what are the differences between satellite and katello deployments, to me it's the same thing except branding. But if we need to differentiate between them, we should totally use just `katello` and `satellite` attributes or both.

In the PR you refer to, I see nesting of ifdefs. I highly disencourage from this, this can lead into issues, we should treat them as disjoint attributes (foreman is special, we could also create `katello_or_satellite` one that would serve the same purpose). Specifically, `katello` cannot be nested into `satellite` and vice versa. This approach sometimes lead to repeating a sentence or a paragraph, but at least we are not copying whole guides. I think this is worth it.

So, I do not understand why would we need to merge this proposal. To me, making attributes including other attributes is dangerous.

**Comment by ekohl** [comment]
> In the PR you refer to, I see nesting of ifdefs. I highly disencourage from this

I did not intend to nest. I intended "this should be shown if katello is present". The further nesting is then if there's anything specific for Satellite, like mention of Red Hat support vs community support within the chapter.

What you didn't mention is that orcharhino also includes katello. So in most cases we'll write `ifdef::katello,satellite,orcharhino[]`. That is very long (and easy to forget).

So in short: I want some definition when the katello plugin is present. This PR's goal is to make that just `katello`.

If we have a specific need for upstream only documentation (that shouldn't make it into satellite nor orcharhino) we can either use `ifndef::satellite,orcharhino[]` or introduce a new `katello-upstream` tag. However, I think the former is good enough.
